modules/storages/app/services/storages/group_folder_properties_sync_service.rb

```diff
@@ -196,19 +201,34 @@ def group_users
   end
 
   def project_folder_permissions(project:)
-    OAuthClientToken
-      .where(oauth_client: @storage.oauth_client, users: project.users)
-      .includes(:user)
-      .each_with_object({
-                          users: { "#{@nextcloud_system_user}": ALL_PERMISSIONS },
-                          groups: { "#{@group}": NO_PERMISSIONS }
-                        }) do |token, permissions|
+    tokens_query = OAuthClientToken
+                 .where(oauth_client: @storage.oauth_client)
+                 .where.not(id: @admin_tokens_query)
+                 .includes(:user)
+    if !(project.public? && Role.non_member.permissions.intersect?(PERMISSIONS_KEYS))
+      tokens_query = tokens_query.where(users: project.users)
+    end
```

Comment:
This made my head spin bit 😅  WDYT about extracting this to a descriptive method? / adding a comment above

```ruby
    if project_is_private_and_non_members_cannot_access_files?
      tokens_query = tokens_query.where(users: project.users)
    end
    # ....
    
     def project_is_private_and_non_members_cannot_access_files?
        !(project.public? && Role.non_member.permissions.intersect?(PERMISSIONS_KEYS))
     end
```

Reply:
Yeah, it is cryptic :smile: I'll fix it.